components/org.wso2.carbon.secvault.securevault/src/main/java/org/wso2/carbon/secvault/securevault/SecureVaultInitializer.java

```diff
@@ -18,168 +18,130 @@
+            SecureVaultConfiguration secureVaultConfiguration = SecureVaultConfigurationProvider
+                    .getInstance().getConfiguration().orElseThrow(() ->
+                            new SecureVaultException("Error occurred when obtaining secure vault configuration. " +
+                                    "Initialise secure vault configuration first by calling " +
+                                    "SecureVaultConfigurationProvider.getInstance().initSecureVaultConfig() method"));
+            if (SecureVaultUtils.isOSGIEnv()) {
+                masterKeyReader = SecureVaultDataHolder.getInstance().getMasterKeyReader()
+                        .orElseThrow(() -> new SecureVaultException("Error occurred when obtaining secure vault " +
+                                "configuration"));
+                secretRepository = SecureVaultDataHolder.getInstance().getSecretRepository()
+                        .orElseThrow(() -> new SecureVaultException("Cannot initialise secure vault " +
+                                "without secret repository"));
+            } else {
+                String masterKeyReaderType = secureVaultConfiguration.getMasterKeyReaderConfig().getType()
+                        .orElseThrow(() -> new SecureVaultException("Master key reader type is not set"));
+                String secretRepositoryType = secureVaultConfiguration.getSecretRepositoryConfig().getType()
+                        .orElseThrow(() -> new SecureVaultException("Secret repository type is not set"));
+                masterKeyReader = createInstance(masterKeyReaderType, MasterKeyReader.class);
+                secretRepository = createInstance(secretRepositoryType, SecretRepository.class);
+                SecureVaultDataHolder.getInstance().setMasterKeyReader(masterKeyReader);
+                SecureVaultDataHolder.getInstance().setSecretRepository(secretRepository);
             }
+            initializeSecureVaultWithConfig(secureVaultConfiguration, masterKeyReader, secretRepository);
+            initialized = true;
         }
-
         logger.debug("Secure Vault initialized successfully");
         return new SecureVaultImpl();
     }
 
-    public Optional<String> getSecureVaultYAMLPath() {
-        return secureVaultYAMLPath;
-    }
-
-    public void setSecureVaultYAMLPath(Optional<String> secureVaultYAMLPath) {
-        this.secureVaultYAMLPath = secureVaultYAMLPath;
-    }
-
-    public Optional<String> getMasterKeyYAMLPath() {
-        return masterKeyYAMLPath;
-    }
-
-    public void setMasterKeyYAMLPath(Optional<String> masterKeyYAMLPath) {
-        this.masterKeyYAMLPath = masterKeyYAMLPath;
-    }
-
-    public Optional<String> getSecretPropertiesPath() {
-        return secretPropertiesPath;
-    }
-
-    public void setSecretPropertiesPath(Optional<String> secretPropertiesPath) {
-        this.secretPropertiesPath = secretPropertiesPath;
+    /**
+     * Initialise secure vault.
```

Comment:
Method has used 'z' and the comment has used 's' in the word "Initialize". In addition comment has not included method parameter descriptions.